fix(docker): pass collectContainerSize parameter to runDockerCustom method

puretension opened the pull request:
## What does this PR do?

Fixes #39084 

This PR fixes an issue where  and  metrics were not being sent to Datadog when  was enabled via autodiscovery configuration.

## Motivation

The  parameter was calculated in the  method but was not being passed to the  method. This caused the size metrics to be ignored regardless of the configuration setting, leading to missing metrics in the Datadog UI.

## Describe how to test this change

1. Configure a Docker container with autodiscovery:
   

2. Run the agent and verify that  and  metrics are now collected and sent to Datadog.

## Additional Notes

- Updated test cases to include the new  parameter
- Added release note documenting the fix
- No breaking changes to existing functionality

puretension committed:
fix(docker): pass collectContainerSize parameter to runDockerCustom method

Fixes #39084

The collectContainerSize parameter was not being passed to the runDockerCustom
method, causing docker.container.size_* metrics to be ignored regardless of
the collect_container_size configuration setting.

Changes:
- Add collectContainerSize parameter to runDockerCustom method signature
- Pass collectContainerSize from Run method to runDockerCustom
- Update size metric collection condition to check collectContainerSize
- Update test calls to include the new parameter

puretension commented on releasenotes/notes/fix-docker-container-size-metrics-39084-5e3ccb6d216ea716.yaml:
```diff
@@ -0,0 +1,9 @@
+    metrics were not being sent to Datadog when ``collect_container_size`` was enabled 
+    via autodiscovery configuration. The ``collectContainerSize`` parameter was not being 
+    properly passed to the ``runDockerCustom`` method, causing the size metrics to be 
+    ignored regardless of the configuration setting.
```
@estherk15 Thank you for the suggestion! 
The revised release note is much clearer and more user-friendly. 
I've applied your changes. 👍

L3n41c reviewed (commented):
Hello @puretension,

Thanks for submitting this contribution!

Could you please review my comment?
The PR description says that it adds a way to **enable** `docker.container.size_{rw,rootfs}` metrics emission whereas the code change looks like it adds a way to **disable** it.

Could you clarify the intent?

puretension committed:
fix: remove redundant collectContainerSize condition in size metrics check

The collectContainerSize parameter already controls Docker API size collection
via Size: collectContainerSize. When false, rawContainer.SizeRw and
rawContainer.SizeRootFs will be 0, making the additional condition redundant.

Signed-off-by: puretension <rlrlfhtm5@gmail.com>

L3n41c reviewed (commented):
Hi again @puretension,

Thank you for your quick update.

I’m wondering why the new `collectContainerSize` parameter has been kept on the `runDockerCustom` function since it’s unused.
